api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -770,58 +770,48 @@ public CohortDefinition getN2() {
     CohortDefinition rapidFlow = getPatientsWithStartOrContinueOnRapidFlow();
 
     String mappings = "onOrBefore=${endDate},location=${location}";
-    //  cd.addSearch("TxCurr", EptsReportUtils.map(txCurr, mappings));
-    //  cd.addSearch("scheduledN2", EptsReportUtils.map(patientsScheduled, mappings));
+    cd.addSearch("TxCurr", EptsReportUtils.map(txCurr, mappings));
+    cd.addSearch("scheduledN2", EptsReportUtils.map(patientsScheduled, mappings));
 
-    String rapidFlowMappings =
-        "onOrAfter=${startDate},onOrBefore=${endDate},locationList=${location}";
     cd.addSearch("rapidFlow", EptsReportUtils.map(rapidFlow, mappings));
-    /* cd.addSearch(
-        "completed", EptsReportUtils.map(getPatientsWhoCompletedRapidFlow(), mappings));
+    cd.addSearch("completed", EptsReportUtils.map(getPatientsWhoCompletedRapidFlow(), mappings));
 
     cd.addSearch(
         "nonPregnantNonBreastFeedingNonTb",
         EptsReportUtils.map(
             getPregnantAndBreastfeedingAndOnTBTreatment(),
             "endDate=${endDate},location=${location}"));
-    /*cd.setCompositionString(
-        "TxCurr AND (scheduledN2 OR rapidFlow) NOT (completed OR nonPregnantNonBreastFeedingNonTb)");*/
-    cd.setCompositionString("rapidFlow");
+    cd.setCompositionString(
+        "TxCurr AND (scheduledN2 OR rapidFlow) NOT (completed OR nonPregnantNonBreastFeedingNonTb)");
```

Comment:
Can that nagation be preededed by AND? instaed of just NOT, let it be **AND NOT**

Reply:
@ningosi in this case i didn't changed that, and in all indicators from N1 to N9 are implemented only with **NOT** operators.

Reply:
That is fine, but in future it explain better by saying **AND NOT** other than just doing **NOT** NOT could be used better if you only have one definition that you need the reverse eg **NOT stable** will give you unstable patients, but we can leave it as it is for now